okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpChannelBuilder.java

```diff
@@ -144,6 +158,19 @@ public OkHttpChannelBuilder connectionSpec(ConnectionSpec connectionSpec) {
   protected ChannelEssentials buildEssentials() {
     final ExecutorService executor = (transportExecutor == null)
         ? SharedResourceHolder.get(DEFAULT_TRANSPORT_THREAD_POOL) : transportExecutor;
+    switch (negotiationType) {
+      case TLS:
+        if (sslSocketFactory == null) {
+          sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+        }
+        break;
+      case PLAINTEXT:
+        sslSocketFactory = null;
```

Comment:
Does it hurt anything if we pass an sslSocketFactory to OkHttpClientTransportFactory?

I'd be fine initializing sslSocketFactory to SSLSocketFactory.getDefault(), except I worry that the method may do a bit too much loading.